start.py

```diff
@@ -127,5 +132,77 @@ def failure(timeout):
+    result_rpms = []
+    sdc_version = whole_sdc_version.rsplit('-RC')[0]
+    sdc_rpm = 'streamsets-datacollector-{}-1.noarch.rpm'.format(sdc_version)
+    result_rpms.append(sdc_rpm)
+    name = 'streamsets-datacollector-mapr_{mapr_version}-lib-{sdc_version}-1.noarch.rpm'
+    sdc_mapr_rpm = name.format(sdc_version=sdc_version,
+                               mapr_version='_'.join(mapr_version.split('.')[:2]))
+    result_rpms.append(sdc_mapr_rpm)
+
+    result_wgets = []
+    el_part = ''
+    if sdc_version.startswith('3'):
+        el_part = 'el7/' if mapr_version.startswith('6.0.0') else 'el6/'
+    base_url = '{sdc_repo}{sdc_ver}/rpm/{el_part}'.format(sdc_repo=DEFAULT_SDC_REPO,
+                                                          sdc_ver=whole_sdc_version,
+                                                          el_part=el_part)
+
+    # RPM for SDC.
+    result_wgets.append('wget -q {base_url}{sdc_rpm}'.format(base_url=base_url, sdc_rpm=sdc_rpm))
+    # RPM for MapR stage library.
+    result_wgets.append('wget -q {base_url}{sdc_mapr_rpm}'.format(base_url=base_url,
+                                                                  sdc_mapr_rpm=sdc_mapr_rpm))
+    # RPM for MEP stage library for MapR 6.
+    if mapr_version.startswith('6.0.0'):
+        mep_rpm_name = 'streamsets-datacollector-mapr_{}-mep{}-lib-{}-1.noarch.rpm'
+        mep_rpm = mep_rpm_name.format('_'.join(mapr_version.split('.')[:2]),
+                                      mep_version[:1],
+                                      sdc_version)
+        result_wgets.append('wget -q {base_url}{mep_rpm}'.format(base_url=base_url,
+                                                                 mep_rpm=mep_rpm))
+        result_rpms.append(mep_rpm)
+    # RPM for Spark 2.1 stage library for MapR 5.2.2.
+    if mapr_version == '5.2.2' and mep_version.startswith('3'):
+        spark_rpm_name = 'streamsets-datacollector-mapr_spark_2_1_mep_{}-lib-{}-1.noarch.rpm'
+        spark_rpm = spark_rpm_name.format('_'.join(mep_version.split('.')[:2]), sdc_version)
+        result_wgets.append('wget -q {base_url}{spark_rpm}'.format(base_url=base_url,
+                                                                   spark_rpm=spark_rpm))
+        result_rpms.append(spark_rpm)
+
+    return result_wgets, result_rpms
+
+
+def _install_streamsets_datacollector(primary_node, sdc_version, mapr_version, mep_version):
+    wget_commands, rpm_names = _gather_wget_commands_and_rpm_names(sdc_version, mapr_version, mep_version)
+    java_opts_cmd = ("sed -i 's|#export SDC_JAVA_OPTS=|export SDC_JAVA_OPTS=|' "
+                     " /opt/streamsets-datacollector/libexec/sdcd-env.sh")
+    commands = ['export JAVA_HOME=/usr/java/jdk1.8.0_131',
+                'cd /opt',
+                'yum -y -q localinstall {}'.format(' '.join(rpm_names)),
+                java_opts_cmd,
+                'export SDC_HOME=/opt/streamsets-datacollector',
```

Comment:
Are these exports just so that things are present when `streamsets setup-mapr` is called? If so, maybe avoid doing them as exports and just do `SDC_CONF=/etc/sdc MAPR_VERSION={} ... /opt/streamsets-datacollector/bin/streamsets setup-mapr` all on one line?

Also, may want to go through and remove hardcoded strings in favor of string literals defined near the top of the file to make it easier to change things if needed.